bpf/bpf_lxc.c

```diff
@@ -679,6 +690,17 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 		}
 	}
 
+#if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_ROUTING)
+	/* If the destination is the local host and per-endpoint routes are
+	 * enabled, jump to the bpf_host program to enforce ingress host policies.
+	 */
+	if (*dstID == HOST_ID) {
+		ctx_store_meta(ctx, CB_FROM_HOST, 0);
+		tail_call_static(ctx, &POLICY_CALL_MAP, HOST_EP_ID);
```

Comment:
Are we OK that we don't return back to the same program? Maybe we should add a big warning saying that if we extend the statements below, we should consider this case which does not return back.

Reply:
Yes, all subsequent code is to handle the packet delivery into the tunnel, the stack, or other devices. Egress processing (ct, policy enforcement, NATing) is finished at this point.